zoo_frontend/src/pages/food/edit/edit.jsx

```diff
@@ -97,53 +94,16 @@ export default class extends Component {
       const budgetCode = await serverFoodAPI.getRelatedBudgetCode(query.id);
 
       const nutData = await serverNutDataAPI.getNutData({ where: { foodId: query.id } });
-      // loop over nutData results and grab their respective nutrition sources
-      const nutDataSourcePromises = nutData.data.map((nut) => new Promise((resolve, reject) => {
-        const nutId = nut.dataId;
-        serverNutDataAPI.getRelatedNutritonSource(nutId).then((res) => {
-          resolve({ ...res.data, dataId: nutId });
-        }, (rej) => {
-          reject(rej);
-        });
-      }));
-
-      const nutDataDataSources = await Promise.all(nutDataSourcePromises).catch((err) => {
-        console.error(err);
-      });
-
-      // loop over nutData results and grab their respective nutrition sources
-      const nutDataNutrDefPromises = nutData.data.map((nut) => new Promise((resolve, reject) => {
-        const nutId = nut.dataId;
-        serverNutDataAPI.getRelatedNutrDef(nutId).then((res) => {
-          resolve({ ...res.data, dataId: nutId });
-        }, (rej) => {
-          reject(rej);
-        });
-      }));
+      const foodWeights = await serverFoodWeightsAPI.getFoodWeight({ where: { foodId: query.id } });
 
-      const nutDataNutrDefs = await Promise.all(nutDataNutrDefPromises).catch((err) => {
+      const allNutrients = await serverNutrDataAPI.getNutrDef().catch((err) => {
         console.error(err);
       });
-
-
-      // loop over nutData results and grab their respective nutrition sources
-      const nutDataSourcedFromPromises = nutData.data.map((nut) => new Promise((resolve, reject) => {
-        const nutId = nut.dataId;
-        serverNutDataAPI.getRelatedSourceCd(nutId).then((res) => {
-          resolve({ ...res.data, dataId: nutId });
-        }, (rej) => {
-          reject(rej);
-        });
-      }));
-
-      const nutDataSourcedFrom = await Promise.all(nutDataSourcedFromPromises).catch((err) => {
+      const allSources = await serverDataSrcAPI.getSources().catch((err) => {
         console.error(err);
       });
 
-      const allNutrients = await serverNutrDataAPI.getNutrDef().catch((err) => {
-        console.error(err);
-      });
-      const allSources = await serverDataSrcAPI.getSources().catch((err) => {
+      const allUnits = await serverUnitsAPI.getUnits().catch((err) => {
```

Comment:
doing const a = await something.doSomething().catch((err) => whatever) does not terminate the function if it fails and it will leave a as undefined then continue the function.